src/imp/schannel.rs

```diff
@@ -63,6 +65,18 @@ pub struct Identity {
     cert: CertContext,
 }
 
+// used for the from_os_provider function
+enum OsProviderParameters {
+    ContextFromStore {
+        store_name: String,
+        is_machine: bool,
+        decoded_hex: Vec<u8>,
+    },
+    ContextFromFile {
+        file_path: PathBuf,
+    },
```

Comment:
I would use unnamed fields here since they are all different types and the enum is only used privately